portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -86,19 +98,57 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   /** Posts a comment retrieved from the form input adding it to the messages variable. */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get the comment from the form and add it to the array.
+    // Get the comment and image from the form and add it to the array.
     String content = request.getParameter("text-input");
+    String imageUrl = getUploadedFileUrl(request, "comment-image");
     long timestamp = System.currentTimeMillis();
 
-    // Create an Entity that holds the comment and the time it was created
-    // and store it in Datastore.
+    // Create an Entity that holds the comment, the image
+    // and the time it was created and store it in Datastore.
     Entity commentEntity = new Entity("Comment");
     commentEntity.setProperty("content", content);
+    commentEntity.setProperty("imageUrl", imageUrl);
     commentEntity.setProperty("timestamp", timestamp);
+
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     // Redirect back to the HTML page.
     response.sendRedirect("/index.html");
   }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // The form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
```

Comment:
Same as above, can you clarify what this comment means? (live server)

Reply:
Those comments were copied from one of the examples in the tutorials. They describe the edge cases that follow them, and I'm not 100% sure of this, but I think live server means when deployed, and dev server means when I run a local server. I modified them to explain this.